pkg/api/util/conditions.go

```diff
@@ -219,3 +219,21 @@ func CertificateRequestHasCondition(cr *cmapi.CertificateRequest, c cmapi.Certif
 	}
 	return false
 }
+
+// This returns the status reason of a CertificateRequest. The order of reason
+// hierarchy is 'Failed' -> 'Ready' -> 'Pending' -> ''
+func CertificateRequestStatusReason(cr *cmapi.CertificateRequest) string {
+	for _, reason := range []string{
+		cmapi.CertificateRequestReasonFailed,
+		cmapi.CertificateRequestReasonIssued,
+		cmapi.CertificateRequestReasonPending,
+	} {
+		for _, con := range cr.Status.Conditions {
```

Comment:
We should only be checking the condition of type `Ready` here - instead we are looping through every condition and if any of them have a matching reason, we return it.

Reply:
Is that right? We also want to know if its in a Failed state for the calling switch statement

Reply:
Failed is a *reason* though, not a *type*. The type that we set in all these cases is `Ready`, and the reason for Ready being False might be `Failed`.

Right now, if we added a new condition named `Foo` and set the reason to `Failed`, this function would match despite the `reason` for `Ready` potentially being something different.

Reply:
:+1: